examples/fluidsim.dx

```diff
@@ -7,15 +7,15 @@ import plot
 
 def wrapidx (n:Type) [Ix n] (i:Int) : n =
   -- Index wrapping around at ends.
-  asidx $ mod i $ size n
+  asidx $ unsafe_i_to_n $ mod i $ n_to_i $ size n
```

Comment:
`mod` of all functions should support  `Nat`
It is a little dubious as to if it should support `Int`.
Mathematically I think it should only support `Nat` and we should introduce something else that has signed behavour and supports `Int`

Reply:
Good point! I assume you mean it should support (or require) `Nat` for its second argument rather than its first. We define `mod` in terms of `rem` from the `Integral` class. Maybe `rem` should have type `rem : [Integral a] -> a -> Nat -> a`, or even `[Integral a] -> a -> Nat -> Nat` or even `[Integral a] -> a -> (n:Nat) -> Fin n` !